rosidl_pycommon/rosidl_pycommon/__init__.py

```diff
@@ -31,7 +32,11 @@
 from rosidl_parser.parser import parse_idl_file
 
 
-def convert_camel_case_to_lower_case_underscore(value):
+if TYPE_CHECKING:
+    from _typeshed import FileDescriptorOrPath
```

Comment:
Looks like rhel and windows are unhappy with this one. Seems like it was added in https://github.com/python/typeshed/pull/9475, maybe the version of `mypy` on those platforms has too old a version of `typeshed`? Maybe we could `try/except` and define the type here on `ImportError`?

```
18:46:38 rosidl_pycommon\__init__.py:36:5: error: Module "_typeshed" has no attribute "FileDescriptorOrPath"; maybe "FileDescriptor"?
```

Reply:
Would you prefer copying the definition out of typeshed on ImportError or just setting to Any on failure?

Reply:
I like that `Any` seems more likely to pass CI the first time. Since I expect this only affects older versions of mypy the easiest solution sounds like the best one to me.

Reply:
Ok I think I got it working with a hasattr check.